### PR TITLE
Patch button variables

### PR DESCRIPTION
**Pull request by kweij**
- Moved button variables to a variables file
- Added and used variables consistently for all types (default, secondary, small, large)
- Removed _lib-button-icon() in favor of lib-button-icon() with defaults

### CODE REVIEW

**Comment by Igloczek** [comment]
Looks like a #142

**Comment by kweij** [comment]
It does indeed - I was working on the same thing when I found out about #142. However, this one has all used settings available as variables. I find it a more thorough solution. Please consider it.

**Comment by Igloczek** [comment]
@kweij Could you check what's wrong after the merge of #142?

File: styles/vendor/magento-ui/variables/_buttons.scss [hunk 1]
```diff
@@ -1,51 +1,98 @@
...
-$button__border                     : 1px solid $color-gray-darken2 !default;
-$button__border-radius              : 3px !default;
-$button__hover__color               : $color-gray-darken3 !default;
-$button__hover__background          : $color-gray-darken1 !default;
-$button__active__background         : $button__color !default;
+//  Button variables
+//  _____________________________________________
+
+//  Structural
+$button__cursor: pointer !default;
```
Comment (Igloczek) [inline styles/vendor/magento-ui/variables/_buttons.scss:6]: Could you align these variables vertically, same as they were aligned before?

File: styles/blocks/_buttons.scss [hunk 2]
```diff
@@ -1,12 +1,3 @@
-a.action.primary,
-button {
-    border-radius: 3px;
-}
-button {
-    &:active {
-        box-shadow: $button__shadow;
-    }
-}
```
Comment (Igloczek) [inline styles/blocks/_buttons.scss:9]: Why is this code removed? It look like in the core - https://github.com/magento/magento2/blob/develop/app/design/frontend/Magento/blank/web/css/source/_buttons.less

**Review by Igloczek — CHANGES_REQUESTED** [review]

**Comment by Igloczek** [comment]
The content of this PR seems to be not valid anymore, b/c of the changes introduced in v1.0.0.

If you think it's still a case, please read the new contribution rules first.

**Pull request closed without merging**